web/app/view/StateController.js

```diff
@@ -184,5 +184,30 @@ Ext.define('Traccar.view.StateController', {
             this.position = null;
             Ext.getStore('Attributes').removeAll();
         }
+    },
+
+    onCellClick: function (grid, td, cellIndex, record, tr, rowIndex, e) {
+        var positionId = this.position.getId();
+        if (record.get('name') === Strings.positionAddress && !record.get('value') &&
+                e.target.tagName.toLowerCase() === 'a') {
+            Ext.Ajax.request({
+                scope: this,
+                method: 'GET',
+                url: 'api/server/geocode',
+                params: {
+                    latitude: this.position.get('latitude'),
+                    longitude: this.position.get('longitude')
+                },
+                success: function (response) {
+                    if (record && this.position && positionId === this.position.getId()) {
```

Comment:
This is protection in case request completed after user clicked other position or deselected position.